src/Sarif/Baseline/ResultMatching/SarifLogMatcher.cs

```diff
@@ -239,22 +239,28 @@ private Rule GetRuleFromResources(Result result, IDictionary<string, Rule> rules
                 properties = currentRuns.Last().Properties;
             }
 
+            var remappedLogicalLocations = new Dictionary<LogicalLocation, int>();
+
             properties = properties ?? new Dictionary<string, SerializedPropertyInfo>();
 
             List<Result> newRunResults = new List<Result>();
             foreach (MatchedResults resultPair in results)
             {
-                newRunResults.Add(resultPair.CalculateNewBaselineResult(PropertyBagMergeBehavior));
+                Result result = resultPair.CalculateBasedlinedResult(PropertyBagMergeBehavior);
+                newRunResults.Add(resultPair.CalculateBasedlinedResult(PropertyBagMergeBehavior));
+
+                var logicalLocationIndexRemapper = new LogicalLocationIndexRemapper(resultPair.LogicalLocations, remappedLogicalLocations);
```

Comment:
>LogicalLocationIndexRemapper [](start = 55, length = 28)

this visitor fixes up indices so that merged results (absent, shared and new) all have logical location index property values that correspond to the newly merged logical locations in the computed baseline.